week-2-wide-perspective/w02e02-data-manipulation/src/observers/ordersObserver.js

```diff
@@ -0,0 +1,80 @@
+import { addOrder, getOrders } from "../services/http.service.js";
+
+export class Observable {
+    constructor() {
+        this.observers = [];
+
+    }
+    addObserver(observer) {
+        this.observers.push(observer);
+    }
+
+
+    removeObserver(observer) {
+        const removeIndex = this.observers.indexOf(observer);
+        if (removeIndex !== -1) {
+            this.observers = this.observers.slice(removeIndex, 1);
+        }
+    }
+
+
+    notify(data) {
+        console.log('notify')
+        if (this.observers.length > 0) {
+            this.observers.forEach((observer) => observer.update(data))
+        }
+    }
+}
+
+export class Observer {
+    // Gets called by the Subject::notify method.
+    update() { }
+}
+export class DataTile extends Observer {
```

Comment:
To podejście jako jedyne zmieniłbym całkowicie.

Pomijając to że tutaj masz całą masę klas które można by wydzielić do osobnych plików (o czym pisałem wyżej).

To tutaj robisz `Tight cupling` swojego observable rozwiązania z widokiem.
Przecież można odwrotnie.

To każdy komponent z `/tiles` powinien mieć subskrypcję:

przykładowo jesteś wewnątrz `/tiles/BestSaleTile.js`:

```javascript
content.innerHTML = '5522.0'
```

zamiana na:

```javascript
ordersState.addObserver({
   update: (state) => {
      content.innerHTML = // użyj state
   }
 })
```

Wtedy to sam koponent dostaje "suche" dane od `Observera` i nie musisz się bawić w osobne determinowanie `dataToDisplay`
ponieważ to sam komponent będzie decydował jak użyje i jak odświeży swój widok na bazie danych ze stanu.

widzisz to?